app/jobs/cleanup_account_job.rb

```diff
@@ -1,18 +1,27 @@
 class CleanupAccountJob < ActiveJob::Base
   def perform(account)
-    cleanup_solr(account)
-    cleanup_fedora(account)
+    account.switch do
+      cleanup_solr(account)
+      cleanup_fedora(account)
+      # Redis doesn't need account, as account.switch changes Redis to correct namespace
+      cleanup_redis
+    end
     Apartment::Tenant.drop(account.tenant)
     account.destroy
   end
 
   private
 
     def cleanup_fedora(account)
-      account.switch do
-        # Preceding slash must be removed from base_path when calling delete()
-        fcrepo_client.delete(account.fcrepo_endpoint.base_path.sub!(%r{^\/}, ''))
-      end
+      # Preceding slash must be removed from base_path when calling delete()
+      fcrepo_client.delete(account.fcrepo_endpoint.base_path.sub!(%r{^\/}, ''))
+    end
+
+    def cleanup_redis
+      # Redis::Namespace currently doesn't support flushall or flushdb.
+      # So, we must loop through and delete all keys in this namespace.
+      # See https://github.com/resque/redis-namespace/issues/56
+      redis_namespace.keys.map { |key| redis_namespace.del(key) }
```

Comment:
Thanks for the notes. Agreed, a bulk delete via asterisk may be better. I'll look into whether that works properly for `Redis::Namespace` today. The `scan_each` option would also work, though since we have a `Redis::Namespace` (limited by current namespace) already, we wouldn't need to match against the `namespace` a second time.